projects/packages/publicize/src/class-publicize.php

```diff
@@ -630,6 +631,16 @@ public function post_is_done_sharing( $post_id = null ) {
 		return 'publish' === $post->post_status || get_post_meta( $post->ID, $this->POST_DONE . 'all', true ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 	}
 
+	/**
+	 * Returns whether or not the nudge to show the upgrade to Jetpack Social Plan.
+	 *
+	 * @return bool True if the nudge has to be shown, false otherwise.
+	 */
+	public function should_show_social_plan_nudge() {
+		Current_Plan::refresh_from_wpcom();
```

Comment:
refresh_from_wpcom internally calls the `/sites/%d'` end-point which will always give us the latest plans and feature info. 

in the Jetpack plugin, Jetpack calls `refresh_from_wpcom` on every heartbeat, which happens only every 24 hours, however, the plan is also refreshed on every page refresh within the Jetpack Plugin by calling https://opengrok.a8c.com/source/xref/jetpack/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php?r=dc2d81eb#1916:~:text=1916,(%20%24response%20)%3B


So I decided it's safe to call `refresh_from_wpcom` on every page load within the Jetpack Social plugin as well.